tree/tree.h

```diff
@@ -2,46 +2,81 @@
 #define  _MACHINELEARNING_TREE_TREE_H__
 
 #include "splitter.h"
+//#include "criterion.h"
 
 #include <vector>
 
 namespace tree {
 
 /**
- * Base class for all Decision Tree implementations.
+ * Decision Tree implementation. It will not be used directly.
+ * This class will be extended by Classifiers and Regressors.
+ * \code
+ *    // creates train and test collections, splitter and criterion pointers...
+ *
+ *    // default value for the number of features drawn randomly.
+ *    float sqrt = sqrt(Dtrain->GetNumberOfFetures())/Dtrain->GetNumberOfFetures(); 
+ *    Tree tree(splitter, criterion, sqrt, 0, 0, 2, 2);
+ *    tree->Build(Dtrain);
+ *    std::vector<Node *> leafs = new std::vector<Node *>(); 
+ *    tree->Traverse(Dtest, leafs);
+ *
+ *    // free pointers...
+ * \endcode
  */
 class Tree{
 
   public:
 
     /**
-     * \brief Standard destructor.
+     * Tree class constructor.
+     * \param splitter          Splitter object (It doesn't have ownership).
+     * \param criterion         Criterion object (It doesn't have ownership).
+     * \param max_features      The maximum number of features used to create the tree.
+     * \param max_depth         The maximum depth of the tree.
+     * \param max_leafs         The maximum number of leaf nodes of the tree.
+     * \param min_samples_leaf  The minimum number of samples per leaf.
+     * \param min_samples_split The minimum number of samples per node in order to be splitted in two.
      */
-    virtual ~Tree();
+    Tree(Splitter *splitter, /*Criterion *criterion,*/ float max_features,
+       int max_depth, int max_leafs, int min_samples_leaf, int min_samples_split);
 
     /**
-     * Builds the Decision Tree
-     * \param X              An object of Collection
-     * \param sample_weight  Weight of each sample in collection
-     * \param feature_weight Weight of each feature 
+     * \brief Standard destructor.
      */
-    virtual void Build(/*Collection X,*/
-                        std::vector<double> &sample_weight, std::vector<double> &feature_weight) = 0;
+    ~Tree();
 
     /**
-     * Predicts the labels of the instances of X based on the fitted model
-     * \return Instance labels
+     * Builds the Decision Tree.
+     * \param X        An object of Collection.
+     * \param sample_weight  Weight of each sample in collection.
+     * \param feature_weight Weight of each feature .
      */
-    virtual std::vector<int> &Predict(/*Collectiin X*/) = 0;
+    void Build(/*const Collection *X,*/
+          const std::vector<double> &sample_weight, const std::vector<double> &feature_weight);
 
+    
     /**
-     *  Returns the predicted probability distribution of each instance in X
-     *  over all possible classes.
-     *  \return Instace probability distribution
+     * Returns the leafs which each instance belongs to.
+     * \param X     An object of Collection.
```

Comment:
We will have to change the name of this parameter, but lets do this in a later PR.